### ISSUE TITLE
Should no-redeclare trigger for function locals?

### ISSUE THREAD

**Issue by hekto** [eslint/eslint#4560]
## test.js
```
/* eslint-env browser, es6 */
/* global win:false, obj:false */
/* eslint no-undef:2, no-redeclare: [2, {builtinGlobals: true}], no-shadow:2 */

"use strict";

const Object = 1;
const obj = 1;

function f() {
    const window = 1;
    const win = 1;
}
```

## Version
```
$ eslint --version
v1.10.1
```

## Actual
```
$ eslint --no-eslintrc test.js

test.js
  7:7  error  "Object" is already defined  no-redeclare
  8:7  error  "obj" is already defined     no-redeclare

✖ 2 problems (2 errors, 0 warnings)
```

## Expected
Errors for `window` and `win` as well.

**Comment by nzakas** [comment]
This is the correct behavior. `window` and `win` are shadows, not redeclarations.

**Comment by hekto** [comment]
Shouldn't no-shadow trigger then?

**Comment by hekto** [comment]
Nevermind, `no-shadow: [2, {builtinGlobals: true}]` gives the behavior i'm looking for.
Thanks!
